### PR TITLE
Added libUUID to Glibc module map

### PR DESCRIPTION
**Pull request by colemancda**
Its a dependency for Foundation and already a dependency for the Swift
Compiler. Also part of Linux and POSIX. 

Necessary for https://bugs.swift.org/browse/SR-1756 and SE-0069

### CODE REVIEW

**Comment by hpux735** [comment]
Considering uuid is already a dependency for Swift, this seems like a good addition.

File: stdlib/public/Platform/glibc.modulemap.gyb [hunk 1]
```diff
@@ -389,5 +389,10 @@ module SwiftGlibc [system] {
       header "${GLIBC_INCLUDE_PATH}/utime.h"
       export *
     }
+	module uuid {
```
Comment (gribozavr) [inline stdlib/public/Platform/glibc.modulemap.gyb]: Indentation is 2 spaces.

**Comment by gribozavr** [comment]
`uuid.h` is a part of the uuid library, not glibc, so it would not be correct to put it into the glibc module.

Could you explain how https://github.com/apple/swift-corelibs-foundation/pull/417 depends on this PR?  I looked at it, and the diff only includes some naming changes and a test as far as I see.

**Comment by colemancda** [comment]
@gribozavr There is currently no way to import this POSIX library without the Swift Package Manager. While this works fine for third party code, the new `struct UUID` in Swift 3.0 in Foundation requires it to be compatible with the version already released for Darwin in https://github.com/apple/swift/blob/d68c6dcd6b4be19cff9f6e82e2811884de56b6fa/stdlib/public/SDK/Foundation/UUID.swift

As you can read here https://bugs.swift.org/browse/SR-1756 there defined is no way to import this library in Swift right now. I believe this would solve that problem. Maybe it does not belong in Glibc, but it should be able to me importable in playgrounds and such a and not be a third party module.

File: stdlib/public/Platform/glibc.modulemap.gyb [hunk 2]
```diff
@@ -389,5 +389,10 @@ module SwiftGlibc [system] {
...
+	  header "{GLIBC_INCLUDE_PATH}/uuid/uuid.h"
```
Comment (jrose-apple) [inline stdlib/public/Platform/glibc.modulemap.gyb]: Missing a dollar sign here.

**Comment by gribozavr** [comment]
@colemancda Could you move the header into `module CUUID [system] {`?

**Comment by gribozavr** [comment]
@colemancda Could you also squash the history to remove the unimportant merges?

**Comment by colemancda** [comment]
Sure, give me a min.

	Coleman,

**Comment by colemancda** [comment]
> @colemancda Could you move the header into module CUUID [system] {?
@gribozavr Same file in Glibc? Or another module map?

**Comment by gribozavr** [comment]
Same file would be fine.

**Comment by colemancda** [comment]
Gonna squash commits now. Please check that the module map is acceptable.

**Comment by gribozavr** [comment]
@colemancda Indentation should be 2 spaces.

Would you mind also adding a simple test to make sure the module map continues working going forward?

**Comment by colemancda** [comment]
Ok, fixed the indentation, gonna add a test. Where would be ideal for me to put it in?

**Comment by gribozavr** [comment]
`validation-test/stdlib/CUUID.swift` I think would be fine.

**Comment by colemancda** [comment]
@gribozavr Added the validation test, could you check it please. Once you think everything is good I'll squash the commits.

**Comment by colemancda** [comment]
Cleaner version at #3107

**Pull request closed without merging**